src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -819,6 +819,7 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <Target Name="PublishItemsOutputGroup" DependsOnTargets="$(PublishItemsOutputGroupDependsOn)" Returns="@(PublishItemsOutputGroupOutputs)">
     <ItemGroup>
+      <PublishItemsOutputGroupOutputs Include="$(ProjectDepsFilePath)" TargetPath="$(ProjectDepsFileName)" />
```

Comment:
So `$(ProjectDepsFilePath)` will be the deps file from the build output, which currently may differ from the one that might get generated from the publish operation.  We're planning on hopefully making these one and the same for 3.0, but right now there's a possibility that the publish targets might [generate its own deps file in the publish directory](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L728).

If a deps file does get generated on publish, then the path would be in [this property](https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets#L738).

However, this target gets called last in publishing and occurs after the calculation for resolved files to publish.  We should refactor is so that the property gets calculated earlier.  If it's empty, you could use `$(ProjectDepsFilePath)` instead.

Reply:
I've updated it to use PublishDepsFilePath instead of ProjectDepsFilePath if PublishDepsFilePath is set.  However, I don't think making this output group depend on the target that calculates PublishDepsFilePath (GeneratePublishDependencyFile) since this output group will be used for design time builds.  In fact, making that change results in the design time build failing and this output group being empty.

Reply:
Right, I think we can safely move setting the property to dependent target would work.  I think moving it to `_ComputeUseBuildDependencyFile` would do the trick, since both `GeneratePublishDependencyFile` and `ComputeFilesToPublish` depend on it.

Reply:
updated